src/main.cpp

```diff
@@ -2,62 +2,400 @@
+            {"plot-raw-input", optional_argument, 0, plotRawInput},
+            {"plot-input", optional_argument, 0, plotInput},
+            {"plot-output", optional_argument, 0, plotOutput},
+            {"print-centroids", optional_argument, 0, printCentroids},
+            {"compare-centroids", optional_argument, 0, compareCentroids},
+            {"compare-stars", optional_argument, 0, compareStars},
+            {"print-attitude", optional_argument, 0, printAttitude},
+            {"compareAttitude", optional_argument, 0, compareAttitude},
+            {"help",            no_argument, 0, help},
+            {0, 0, 0, 0}
+        };
+
+        lost::PipelineOptions pipelineOptions;
+        int index;
+        int option;
+        
+        while (optind < argc) {
+            if (optind < 2) optind++;
+
+            if ((option = getopt_long(argc, argv, "", long_options, &index)) != -1) {
+                switch (option) {
+                    case png :
+                        std::cout << "You made the file path " << optarg << std::endl;
+                        pipelineOptions.png = optarg;
+                        break;
+                    case focalLength :
+                        std::cout << "You set the focal length to " << optarg << std::endl;
+                        pipelineOptions.focalLength = atof(optarg);
+                        break;
+                    case pixelSize :
+                        std::cout << "You set the pixel size to " << optarg << std::endl;
+                        pipelineOptions.pixelSize = atof(optarg);
+                        break;
+                    case fov :
+                        std::cout << "You set the fov to " << optarg << std::endl;
+                        pipelineOptions.fov = atof(optarg);
+                        break;
+                    case centroidAlgo :
+                    {
+                        std::string algo (optarg);
+                        if (algo != "dummy" && algo != "cog" && algo != "iwcog") {
+                            std::cout << "Unrecognized centroid algorithm!" << std::endl;
+                        } else {
+                            std::cout << algo << "!" << std::endl;
+                            pipelineOptions.centroidAlgo = algo;
+                        }                        
+                        break;
+                    }
+                    case centroidDummyStars : 
+                        pipelineOptions.dummyCentroidNumStars = atoi(optarg);
+                        break;
+                    case centroidMagFilter :
+                        std::cout << "You set the centroid mag filter to " << optarg << std::endl;
+                        pipelineOptions.centroidMagFilter = atoi(optarg);
+                        break;
+                    case idAlgo :
+                    {
+                        std::string algo (optarg);
+                        if (algo != "dummy" && algo != "gv" && algo != "pyramid") {
```

Comment:
Not the right place for this check. Instead, use else if statement I described in earlier comment.

Reply:
I don't see an earlier comment?

Reply:
Oops, I must have forgotten to submit it originally. I added another comment just now.